src/lib/components/Pagination.svelte

```diff
@@ -0,0 +1,140 @@
+<script context="module" lang="ts">
+	const PAGE_SIZE = 50
+
+	export function paginate<T extends any[]>(pageParam: string | null, items: T) {
+		const number = pageParam ? parseInt(pageParam) : 1
+		const itemIndexStart = (number - 1) * PAGE_SIZE
+		return {
+			number,
+			items: items.slice(itemIndexStart, itemIndexStart + PAGE_SIZE) as T,
+		}
+	}
+</script>
+
+<script lang="ts">
+	import { goto } from '$app/navigation'
+
+	export let current: number = 1
+	export let results: number
+
+	$: totalPages = Math.ceil(results / PAGE_SIZE)
+	$: buttons = createPageButtons(current, totalPages)
+
+	function createPageButtons(current: number, total: number) {
+		const buttons: (number | null)[] = []
+		if (total === 1) return buttons
+		const startPage = Math.max(1, Math.min(total - 4, current - 2))
+		const endPage = Math.min(total, startPage + 4)
+		for (let p = startPage; p <= endPage; p++) {
+			buttons.push(p)
+		}
+		if (startPage > 1) {
+			if (startPage - 1 > 1) buttons.unshift(null)
+			buttons.unshift(1)
+		}
+		if (endPage < total) {
+			if (endPage < total - 1) buttons.push(null)
+			buttons.push(total)
+		}
+		return buttons
+	}
+</script>
+
+<ul>
+	<li>{results} results</li>
+	{#if current > 1}
+		<li>
+			<a
+				on:click|preventDefault={() => goto(`?page=${current - 1}`)}
+				href="?page={current - 1}"
```

Comment:
Can't we just use the `href` attribute instead of manually calling `goto`? I thought SvelteKit did some sort of JS magic to prevent page reloads when navigating within the site itself.

Reply:
You're right, and I'm not sure why I thought it wasn't already doing that.